controlplane/internal/kubevirt/dns_proxy.go

```
@@ -0,0 +1,340 @@
+	cm := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: TenantDNSCorefileName, Namespace: namespace}}
+	if _, err := controllerutil.CreateOrUpdate(ctx, c, cm, func() error {
+		cm.Data = cmData
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to ensure corefile: %w", err)
+	}
+
+	// 2. Deployment (port 5353 - no SCC needed)
+	deploy := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: TenantDNSDeploymentName, Namespace: namespace}}
+	if _, err := controllerutil.CreateOrUpdate(ctx, c, deploy, func() error {
+		labels := map[string]string{"app": "tenant-dns"}
```

golangci-lint (2.13.2)

[error] string `app` has 4 occurrences, make it a constant

(goconst)